### PR TITLE
[ISSUE #4945] Adjust the storedir format of test cases to the Windows platform

### LINKED ISSUE #4945: Adjust the storedir format of test cases to the Windows platform
image
As shown in the figure above, run on the windows platform baseDir is:
```
C:\Users\LIUJIA~1\AppData\Local\Temp\\unitteststore-df10f848-30cd-4b84-b483-0b53697c96b2
```
image

### PR DESCRIPTION
**Pull request by mxsm**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

close #4945 

## Brief changelog

- Adjust the storedir format of test cases to the Windows platform

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: store/src/test/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHATest.java [hunk 1]
```diff
@@ -72,9 +73,9 @@ public class AutoSwitchHATest {
     private String store2HaAddress;
 
     private BrokerStatsManager brokerStatsManager = new BrokerStatsManager("simpleTest", true);
-    private String tmpdir = System.getProperty("java.io.tmpdir");
-    private String storePathRootParentDir = (StringUtils.endsWith(tmpdir, File.separator) ? tmpdir : tmpdir + File.separator) + UUID.randomUUID();
-    private String storePathRootDir = storePathRootParentDir + File.separator + "store";
+
```
Comment (Oliverwqcwrw) [inline store/src/test/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHATest.java:76]: CI is not passed
Tests in error: 
  AutoSwitchHATest.testTruncateEpochLogAndChangeMaster:380->checkMessage:193 » ConditionTimeout

Tests run: 148, Failures: 0, Errors: 1, Skipped: 9

Comment (mxsm) [reply store/src/test/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHATest.java:76]: I have will fixed it

**Review by Oliverwqcwrw — DISMISSED** [review]
LGTM

**Review by li-xiao-shuang — APPROVED** [review]

**Review by tianliuliu — DISMISSED** [review]

**Comment by ltamber** [comment]
why we don't use `import org.junit.rules.TemporaryFolder`

**Review by tsunghanjacktsai — APPROVED** [review]

**Review by caigy — APPROVED** [review]

**Review by lizhanhui — CHANGES_REQUESTED** [review]
See https://github.com/apache/rocketmq/pull/4957#issuecomment-1235072959

**Comment by mxsm** [comment]
> See [#4957 (comment)](https://github.com/apache/rocketmq/pull/4957#issuecomment-1235072959)

I will do

**Review by ni-ze — DISMISSED** [review]

**Comment by mxsm** [comment]
> Hello @mxsm , There are some conflicts that need to be resolved

I will do

**Review by tsunghanjacktsai — APPROVED** [review]

**Pull request closed without merging**